### ISSUE TITLE
Pass SuffixTable through pipeline for name change heuristics in post-processing

### ISSUE THREAD

**Issue by daniel-j-h** [Project-OSRM/osrm-backend#4201]
After debugging https://github.com/Project-OSRM/osrm-backend/pull/4200 with @MoKob we found out that the `isUTurn` check in post-processing fails for suffix changes (such as `X St North -> X St South`) [here](https://github.com/Project-OSRM/osrm-backend/blob/dd8f5ac01dc38648104f27618e4e4ee5ca33f171/src/engine/guidance/collapse_scenario_detection.cpp#L167) because we don't have our [SuffixTable](https://github.com/Project-OSRM/osrm-backend/blob/dd8f5ac01dc38648104f27618e4e4ee5ca33f171/include/extractor/suffix_table.hpp) available for post-processing.

At the moment the `SuffixTable` is read from the Lua profiles and then used during pre-processing only.

What needs to be done here is:
- [ ] Serialize out the SuffixTable in osrm-extract
- [ ] Implement SuffixTable loading and deserialization in the data facades
- [ ] Pass it into post-processing and make all name-related heuristics suffix-aware

**Comment by emiltin** [comment]
there might be other profile settings that are needed after extraction? it that case maybe write out a file with all these settings, instead a file for each setting?

**Comment by daniel-j-h** [comment]
Yeah we already have the `ProfileProperties` which store global profile data throughout the pipeline.

**Comment by emiltin** [comment]
doesn't that already contain the suffix list then?

**Comment by daniel-j-h** [comment]
No https://github.com/Project-OSRM/osrm-backend/blob/master/include/extractor/profile_properties.hpp

**Comment by emiltin** [comment]
ok i see
